### ISSUE TITLE
Memory Leak in new amqp_consume code?

### ISSUE THREAD

**Issue by pstadt** [net-amqp-rabbitmq/net-amqp-rabbitmq#120]
Hi

I have a simple consumer script archiving data into files.
I updated from perl-Net-AMQP-RabbitMQ-1.500000 to perl-Net-AMQP-RabbitMQ-2.100000. Since than I have seen some oom-killer on that long running archiving script. I rolled back to 1.500000 and everything is fine.
Any Idea if this could be introduced by the new amqp_consume code in 2.000000?
Or any other idea?

**Comment by markwellis** [comment]
Could well be. I'll try to look, but time is lacking at the moment.

Would you be able to create a simple test case that has the leak? Maybe try adapting this test to match what you're doing?

https://github.com/net-amqp-rabbitmq/net-amqp-rabbitmq/blob/master/xt/100_transaction_memory_leak.t

If you could that would be greatly appreciated. If not, I will look at it, but can't guarantee when!

Thanks :)

**Comment by pstadt** [comment]
Something like this?

[102_consumer_memory_leak.t.txt](https://github.com/net-amqp-rabbitmq/net-amqp-rabbitmq/files/309490/102_consumer_memory_leak.t.txt)

Works fine on 1.500000 and collects 4 MB of memory on 2.100001

```
xt/102_consumer_memory_leak.t .. # this test is slow, and probably only works on linux
xt/102_consumer_memory_leak.t .. 1/11 # 0 - used: 6.14mb, diff: 0.02mb
# 10000 - used: 6.46mb, diff: 0.49mb
# 20000 - used: 6.85mb, diff: 0.94mb
# 30000 - used: 7.48mb, diff: 1.41mb
# 40000 - used: 7.86mb, diff: 1.88mb
# 50000 - used: 8.44mb, diff: 2.33mb
# 60000 - used: 8.84mb, diff: 2.84mb
# 70000 - used: 9.41mb, diff: 3.30mb
# 80000 - used: 9.71mb, diff: 3.81mb
# 90000 - used: 10.23mb, diff: 4.30mb
xt/102_consumer_memory_leak.t .. 11/11 
#   Failed test 'memory usage hasn't risen by more than 1mb (4.6171875mb)'
#   at xt/102_consumer_memory_leak.t line 49.
# Looks like you failed 1 test of 11.
xt/102_consumer_memory_leak.t .. Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/11 subtests 
```

**Comment by markwellis** [comment]
Looks perfect. I'll try to look as soon as I can, but it wont be fixed
until the end of the week at the earliest

**Comment by markwellis** [comment]
Fix released to cpan as version 2.200000
